x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -821,16 +824,16 @@ public OpenJobPersistentTasksExecutor(Settings settings, ClusterService clusterS
 
         @Override
         public PersistentTasksCustomMetaData.Assignment getAssignment(OpenJobAction.JobParams params, ClusterState clusterState) {
-            Job foundJob = params.getJob();
-            if (foundJob == null) {
-                // The job was added to the persistent task parameters in 6.6.0
-                // if the field is not present the task was created before 6.6.0.
-                // In which case the job should still be in the clusterstate
-                foundJob = MlMetadata.getMlMetadata(clusterState).getJobs().get(params.getJobId());
+            // If the task parameters do not have a job field then the job
+            // was first opened on a pre v6.6 node and has not been migrated
+
+            // TODO is eligible for migration check (all nodes are at the same version)
+            if (params.getJob() == null) {
+                return AWAITING_MIGRATION;
```

Comment:
The major difference between this and 6.6 is that in a mixed cluster an open job will not get an assignment i.e. a  job left open during rolling upgrade will not be assigned until it has been migrated and the persistent task parameters changed which will only happen once all nodes in the cluster are at least version 6.6.0. 6.7 is capable of running un-migrated jobs so we could allow the un-migrated job to run in a mixed cluster depending on the result of `MlConfigMigrationEligibilityCheck.jobIsEligibleForMigration`.

Either way I need to remove the TODO